### DISCUSSION TITLE
Move back/forth for videos on Android TV (with a remote)

### DISCUSSION THREAD

**Discussion by linosgian** [Radiokot/photoprism-android-client#72] [Ideas]
First off, thanks @Radiokot for the app you've made, it's exactly what I was looking for to browse photos on my phone/Android TV!
I've opened a few issues around things I've found with using it but for the most part the app is working very well.

When you browse videos on an Android TV, if you press the left/right buttons on the remote, instead of moving back/forth in the video, you move to the next/previous video instead, which makes sense, but there is no way for a user with a remote to actually go back to a certain point in a video with a remote. I supposed that If I used a mouse, this wouldn't be a problem (same way it's not for an android phone).

I'm thinking that, this could be done in the same way the official web app does it: you browse with the left/right buttons, and you can press the middle/OK button if you want to watch a video. Once you do that, you can press left/right to move back/forth in the video. If you want to continue to browse, you can press the back button.

This might be suboptimal for mobile users though, I'm not sure.

**Comment by Radiokot** [comment]
Hi. Did you try moving the focus down? The vertical focus order for the video, at least how I tested it on Android TV emulator, should be the following:
1. Screen itself – pressing "OK" toggles controls visibility
2. Play/pause button – pressing "OK" toggles playback state
3. Timeline – Pressing "OK" does nothing, pressing "Left"/"Right" moves the video position
4. Buttons – Pressing "OK" invokes the focused button, pressing "Left"/"Right" moves the focus among buttons

**Reply by Radiokot** [reply]
I mean moving down using the "Up" and "Down" buttons on the remote control

**Reply by linosgian** [reply]
Indeed! It's a little bit difficult to figure where the focus is at any given time (between timeline and the library), but it definitely works, thanks!
